ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -838,18 +838,24 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
    */
 
   public func downloadKeyboard(withID: String, languageID: String, isUpdate: Bool, fetchRepositoryIfNeeded: Bool = true) {
+    let kbdFullID = FullKeyboardID(keyboardID: withID, languageID: languageID)
+    let completionBlock = ResourceDownloadManager.shared.standardKeyboardInstallCompletionBlock(forFullID: kbdFullID, withModel: true)
     ResourceDownloadManager.shared.downloadKeyboard(withID: withID,
                                                     languageID: languageID,
                                                     isUpdate: isUpdate,
-                                                    fetchRepositoryIfNeeded: fetchRepositoryIfNeeded)
+                                                    fetchRepositoryIfNeeded: fetchRepositoryIfNeeded,
+                                                    completionBlock: completionBlock)
```

Comment:
Examining this function in retrospect, I'm not sure if functioned properly before.  The code that would ensure that the downloaded keyboard was added to the user's installed keyboard list wasn't included then.